### PR TITLE
Core, Spark3.5: Fix tests failure due to timeout

### PR DESCRIPTION
**Pull request by manuzhang**
This PR attempts to fix following tests failure due to timeout.

#11047
#11066 
#11651

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java [hunk 1]
```diff
@@ -446,7 +446,7 @@ public void testConcurrentFastAppends(@TempDir File dir) throws Exception {
                 final int currentFilesCount = numCommittedFiles;
                 Awaitility.await()
                     .pollInterval(Duration.ofMillis(10))
-                    .atMost(Duration.ofSeconds(10))
+                    .atMost(Duration.ofSeconds(20))
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java:449]: I'm not sure increasing the timeout is the right thing to do here. Isn't this an issue due to exponential retries during concurrent commits? I'd probably rather define/decrease https://github.com/apache/iceberg/blob/ea5da1789f3eb80fc3196bcdd787a95ac0c493ba/core/src/main/java/org/apache/iceberg/TableProperties.java#L89-L92 for this test

Comment (manuzhang) [reply core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java:449]: Good point. I'd also increase the number of retries.

File: core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java [hunk 2]
```diff
@@ -421,7 +422,7 @@ public void testConcurrentFastAppends(@TempDir File dir) throws Exception {
         TABLES.create(
             SCHEMA,
             SPEC,
-            ImmutableMap.of(COMMIT_NUM_RETRIES, String.valueOf(threadsCount)),
+            ImmutableMap.of(COMMIT_NUM_RETRIES, "10", COMMIT_MIN_RETRY_WAIT_MS, "10"),
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java:425]: why are we updating the number of retries? Also with Hadoop the other component that might affect longer durations is the lock manager's timeout/interval settings: https://github.com/apache/iceberg/blob/34cd01ba2e057866cdb13db8f9919bc98e11e638/core/src/main/java/org/apache/iceberg/CatalogProperties.java#L138-L151

**Comment by manuzhang** [comment]
@nastra Please take another look, thanks!

File: core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java [hunk 3]
```diff
@@ -421,7 +423,13 @@ public void testConcurrentFastAppends(@TempDir File dir) throws Exception {
         TABLES.create(
             SCHEMA,
             SPEC,
-            ImmutableMap.of(COMMIT_NUM_RETRIES, String.valueOf(threadsCount)),
+            ImmutableMap.of(
+                COMMIT_NUM_RETRIES,
+                String.valueOf(threadsCount),
+                COMMIT_MIN_RETRY_WAIT_MS,
+                "10",
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java:430]: Without your changes the test takes 48s for me locally. With your changes it takes 37s. Adding COMMIT_MAX_RETRY_WAIT_MS = "1000" brings it down to 22s, so I would also specify this setting

File: core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java [hunk 4]
```diff
@@ -421,7 +423,13 @@ public void testConcurrentFastAppends(@TempDir File dir) throws Exception {
...
+                LOCK_ACQUIRE_TIMEOUT_MS,
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java:431]: you might want to add a comment here that mentions exactly this

File: core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java (hunk 3 above)
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java:430]: also what about setting LOCK_ACQUIRE_INTERVAL_MS = "100"?

File: spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java [hunk 5]
```diff
@@ -1611,6 +1612,7 @@ public synchronized void testMergeWithSnapshotIsolation()
     sql(
         "ALTER TABLE %s SET TBLPROPERTIES('%s' '%s')",
         tableName, MERGE_ISOLATION_LEVEL, "snapshot");
+    sql("ALTER TABLE %s SET TBLPROPERTIES('%s' '%s')", tableName, COMMIT_MIN_RETRY_WAIT_MS, "10");
```
Comment (nastra) [inline spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java:1615]: this should also set `COMMIT_MAX_RETRY_WAIT_MS`

File: core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java (hunk 3 above)
Comment (manuzhang) [reply core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java:431]: It's not used with `LOCK_ACQUIRE_TIMEOUT_MS=0`

File: spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java [hunk 6]
```diff
@@ -1144,6 +1146,8 @@ public synchronized void testDeleteWithSnapshotIsolation()
     sql(
         "ALTER TABLE %s SET TBLPROPERTIES('%s' '%s')",
         tableName, DELETE_ISOLATION_LEVEL, "snapshot");
+    sql("ALTER TABLE %s SET TBLPROPERTIES('%s' '%s')", tableName, COMMIT_MIN_RETRY_WAIT_MS, "10");
```
Comment (nastra) [inline spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java:1149]: no need to issue multiple ALTER statements. These properties can be set with one ALTER cmd

Comment (manuzhang) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java:1149]: Yes, I just think it will be easier to read.

Comment (nastra) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java:1149]: I don't think we want to issue X ALTER statements just for readability as it increases test duration. This should all happen in a single ALTER stmt

**Comment by nastra** [comment]
@RussellSpitzer or @amogh-jahagirdar can any of you also take a look at this please?

**Review by nastra — APPROVED** [review]

File: spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java [hunk 7]
```diff
@@ -1609,8 +1611,14 @@ public synchronized void testMergeWithSnapshotIsolation()
     createOrReplaceView("source", Collections.singletonList(1), Encoders.INT());
 
     sql(
-        "ALTER TABLE %s SET TBLPROPERTIES('%s' '%s')",
-        tableName, MERGE_ISOLATION_LEVEL, "snapshot");
+        "ALTER TABLE %s SET TBLPROPERTIES('%s', '%s', '%s', '%s', '%s', '%s')",
```
Comment (nastra) [inline spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java:1614]: I think you've got too many commas here. I believe this should be `('%s' '%s', '%s' '%s', '%s' '%s')` and same for the other Spark test

**Review by amogh-jahagirdar — COMMENTED** [review]
So if I understand right, to address flakiness of these tests that are timing out, we're reducing a lot of commit properties (like the max retry and the min retry wait), that way we just fail fast? Typically these should not exceed these values and any tests which do, would just end up failing fast.

Do we have any indication on what could be the root cause for the commits in these tests taking so long? 

To be clear, I'm open to giving the fix in this PR a try but I think we should be a bit cautious that we're not introducing flakiness the other way where another set of tests end up failing more frequently due to not waiting enough for example.

**Comment by nastra** [comment]
@amogh-jahagirdar from what I saw in some local debugging is that these specific tests typically took quite a long time due to exponential retries and eventually timed out, hence this is lowering the min/max wait times for these retries in these 3 particular tests. Other tests shouldn't be affected by this

**Comment by amogh-jahagirdar** [comment]
Sounds good, 

>these specific tests typically took quite a long time due to exponential retries and eventually timed out

Yeah this was the part I was trying to unpack was why are these particular tests retrying a lot? Like I said though, that may be difficult to reason about and we may just want to tune these properties in the interim, so all good!

**Comment by manuzhang** [comment]
@nastra `COMMIT_MAX_RETRY_WAIT_MS='1000'` is not useful when `COMMIT_NUM_RETRIES=4` by default.

**Pull request merged**